Project/Gem/Source/Scripting/PayloadDespawnerComponent.cpp

```diff
@@ -0,0 +1,120 @@
+#include "PayloadDespawnerComponent.h"
+#include "AzCore/Component/ComponentApplicationBus.h"
+#include "AzCore/Debug/Trace.h"
+#include "AzCore/Serialization/EditContext.h"
+#include "AzCore/std/string/string.h"
+#include "AzFramework/Physics/PhysicsSystem.h"
+#include "LmbrCentral/Scripting/TagComponentBus.h"
+#include "ROS2/Frame/ROS2FrameComponent.h"
+#include "ROS2/ROS2Bus.h"
+#include "Scripting/ScriptSpawnSytemBus.h"
+#include "std_msgs/msg/bool.hpp"
+#include <rclcpp/publisher.hpp>
+
+namespace ROS2::Demo
+{
+    void PayloadDespawnerComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<PayloadDespawnerComponent>()->Version(1);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<PayloadDespawnerComponent>("PayloadDespawnerComponent", "PayloadDespawnerComponent")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2::Demo")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"));
+            }
+        }
+    }
+
+    void PayloadDespawnerComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("PhysicsColliderService"));
+    }
+
+    void PayloadDespawnerComponent::Activate()
+    {
+        m_onTriggerEnterHandler = AzPhysics::SimulatedBodyEvents::OnTriggerEnter::Handler(
+            [&]([[maybe_unused]] AzPhysics::SimulatedBodyHandle bodyHandle, [[maybe_unused]] const AzPhysics::TriggerEvent& event)
+            {
+                const auto entityId = event.m_otherBody->GetEntityId();
+                AZ::Entity* entity{};
+                AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+
+                if (IsObjectBox(entityId))
+                {
+                    AZ::Entity* entity{};
+                    AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+                    if (entity)
+                    {
+                        const auto& entityName = entity->GetName();
+                        ScriptSpawnSystemRequestBus::Broadcast(&ScriptSpawnSystemRequestBus::Events::DespawnBox, entityName);
+                    }
+                }
+                else if (IsObjectAMR(entityId))
```

Comment:
I can see slight race conditions here. 
Robot enters the collider first, it updates its cargo status.
It can start to move back, and its payload will be still there. There are two way to address it:
- cache AMR and Boxes to despawn,
- make slight delay sending cargo status (e.g. spin a thread with delay).